### PR TITLE
[toolbars] - new package toolbars-new

### PR DESCRIPTION
**Pull request by noamtwix**

### CODE REVIEW

File: packages/toolbars-new/web/src/Toolbar/buttonsListCreator.js [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+    onSave: 'HEADINGS',
+  },
+  Separator: {
+    type: 'SEPARATOR',
+  },
+  Bold: {
+    icon: BoldIcon,
+    dataHook: 'textInlineStyleButton_BOLD',
+    tooltip: 'Bold',
+    action: 'Bold',
```
Comment (maimonav) [inline packages/toolbars-new/web/src/Toolbar/buttonsListCreator.js:67]:
```suggestion
    action: 'bold',
```

Comment (maimonav) [reply packages/toolbars-new/web/src/Toolbar/buttonsListCreator.js:67]: And for all the rest..:)

File: examples/main/package.json [hunk 2]
```diff
@@ -57,7 +57,8 @@
     "wix-rich-content-plugin-video": "8.47.15",
     "wix-rich-content-preview": "8.47.15",
     "wix-rich-content-text-selection-toolbar": "8.47.15",
-    "wix-rich-content-ui-components": "8.47.15"
+    "wix-rich-content-ui-components": "8.47.15",
+    "wix-rich-content-toolbars-new": "8.47.15"
```
Comment (shaulgo) [inline examples/main/package.json:61]: opt-in

File: examples/main/shared/editor/Editor.tsx [hunk 3]
```diff
@@ -9,12 +9,20 @@ import { TOOLBARS } from 'wix-rich-content-editor-common';
 import {
   DraftContent,
   TextToolbarType,
+  TranslationFunction,
```
Comment (shaulgo) [inline examples/main/shared/editor/Editor.tsx:12]: cleanup file

File: examples/main/shared/editor/EditorPlugins.tsx [hunk 4]
```diff
@@ -729,6 +730,28 @@ const config: RichContentEditorProps['config'] = {
     },
   },
   uiSettings,
+  formattingToolbarButtons: [
```
Comment (shaulgo) [inline examples/main/shared/editor/EditorPlugins.tsx:733]: cleanup

File: packages/common/web/src/types/editorCommandsType.ts [hunk 5]
```diff
@@ -20,6 +20,8 @@ import {
   RICOS_LINK_TYPE,
   RICOS_MENTION_TYPE,
   RICOS_TEXT_HIGHLIGHT_TYPE,
+  RICOS_INDENT_TYPE,
```
Comment (shaulgo) [inline packages/common/web/src/types/editorCommandsType.ts:23]: seprate pr for editor commands

File: packages/editor-common/web/src/Utils/overflowUtils.ts [hunk 6]
```diff
@@ -25,14 +25,39 @@ const getWidth = (element: HTMLElement): number => {
...
     const editorOffset = rootEditor && getElementCoordsInWindow(rootEditor).left;
     const editorWidth = (rootEditor && rootEditor.getBoundingClientRect().width) || 999999;
     if (modalWidth + (modalOffset - editorOffset) > editorWidth) {
+      return modalWidth + (modalOffset - editorOffset) - editorWidth; // optional: return overflow by
+    } else {
+      return false;
+    }
+  } else {
+    console.error('window or document not found'); //eslint-disable-line
+  }
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/overflowUtils.ts:48]: remove safety

File: packages/editor-common/web/src/Utils/overflowUtils.ts [hunk 7]
```diff
@@ -25,14 +25,39 @@ const getWidth = (element: HTMLElement): number => {
   return element.clientWidth;
 };
 
-export const isElementOutOfWindow = (element: HTMLElement): boolean | undefined => {
+const getHeight = (element: HTMLElement): number => {
+  return element.clientHeight;
+};
+
+export const isElementOutOfWindow = (element: HTMLElement): boolean | number | undefined => {
```
Comment (shaulgo) [inline packages/editor-common/web/src/Utils/overflowUtils.ts:32]: 1. new pr
2. elementOverflowWithEditor => {top:number, bottom:....}

File: packages/editor/web/src/RichContentEditor/I18nRichContentEditor.tsx [hunk 8]
```diff
@@ -19,12 +19,22 @@ export default class I18nRichContentEditor extends Component<Partial<RichContent
 
   getToolbarProps = (type: ToolbarType) => this.editor.getToolbarProps(type);
 
+  openMobileAddPlugin = () => this.editor.openMobileAddPlugin();
+
+  getT = () => this.editor.getT();
+
+  getPlugins = () => this.editor.getPlugins();
+
+  removeToolbarFocus = () => this.editor.removeToolbarFocus();
```
Comment (shaulgo) [inline packages/editor/web/src/RichContentEditor/I18nRichContentEditor.tsx:28]: not needed for formatting

**Review by shaulgo — CHANGES_REQUESTED** [review]

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 9]
```diff
@@ -24,9 +24,11 @@ import {
   EditorEventsContext,
   EditorEvents,
 } from 'wix-rich-content-editor-common/libs/EditorEventsContext';
-import { ToolbarType, Version, RicosTranslate } from 'wix-rich-content-common';
+import { ToolbarType, Version, RicosTranslate, EditorCommands } from 'wix-rich-content-common';
 import { emptyDraftContent, getEditorContentSummary } from 'wix-rich-content-editor-common';
 import englishResources from 'wix-rich-content-common/dist/statics/locale/messages_en.json';
+import TextFormattingToolbar from './toolbars/TextFormattingToolbar';
+import LinkToolbar from './toolbars/LinkToolbar';
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/RicosEditor.tsx:31]: dynamic import

Comment (shaulgo) [reply packages/ricos-editor/web/src/RicosEditor.tsx:31]: search for import(

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/editor/web/src/RichContentEditor/I18nRichContentEditor.tsx [hunk 10]
```diff
@@ -19,12 +19,18 @@ export default class I18nRichContentEditor extends Component<Partial<RichContent
...
+
+  getPlugins = () => this.editor.getPlugins();
+
   focus = () => this.editor.focus();
 
   blur = () => this.editor.blur();
 
   getEditorCommands = () => this.editor.EditorCommands;
 
+  getInnerRCERenderedIn = () => this.editor.getInnerRCERenderedIn();
```
[the remark this answers is not in the document]
Comment (noamtwix) [reply packages/editor/web/src/RichContentEditor/I18nRichContentEditor.tsx]: fixed

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 11]
```diff
@@ -253,13 +281,15 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
     this.editor = ref;
   };
 
-  setEditorAndStaticToolbar = (ref: RichContentEditor) => {
+  setEditor = (ref: RichContentEditor) => {
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/RicosEditor.tsx:284]: cleanup

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 12]
```diff
@@ -123,6 +140,17 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
     }
   }
 
+  loadToolbar() {
+    if (this.useNewFormattingToolbar) {
+      import(
+        /* webpackChunkName: "./toolbars/TextFormattingToolbar" */
+        './toolbars/TextFormattingToolbar'
+      ).then(textFormattingToolbarModule => {
+        this.setState({ textFormattingToolbarModule });
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/RicosEditor.tsx:149]:
```suggestion
      ).then({default} => {
        this.setState({ TextFormattingToolbar: default });
```

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 13]
```diff
@@ -296,19 +326,93 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
...
+
+    let hideFormattingToolbar; //when LinkToolbar is open in mobile
+    if (this.useNewFormattingToolbar && activeEditor && isMobile) {
+      const editorCommands: EditorCommands = activeEditor.getEditorCommands();
+      const selection = editorCommands.getSelection();
+      if (isMobile) {
+        hideFormattingToolbar = selection.getIsCollapsed && editorCommands.hasLinkInSelection();
+      } else {
+        hideFormattingToolbar = false;
+      }
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/RicosEditor.tsx:360]:
```suggestion
       hideFormattingToolbar = editorCommands.getSelection().getIsCollapsed && editorCommands.hasLinkInSelection();
```

Comment (shaulgo) [reply packages/ricos-editor/web/src/RicosEditor.tsx:360]: and goes into function

File: packages/ricos-editor/web/src/RicosEditor.tsx [hunk 14]
```diff
@@ -296,19 +326,93 @@ export class RicosEditor extends Component<RicosEditorProps, State> {
...
+    }
+    const onInlineToolbarOpen = toolbarType =>
+      helpers?.onInlineToolbarOpen?.({
+        toolbarType,
+        version: Version.currentVersion,
+      });
+    const onToolbarButtonClick = (
```
Comment (shaulgo) [inline packages/ricos-editor/web/src/RicosEditor.tsx:367]:
```suggestion
    const createOnToolbarButtonClick = (toolbarType) => (
```

**Review by shaulgo — CHANGES_REQUESTED** [review]

File: packages/ricos-editor/web/src/RicosEditor.tsx (hunk 13 above)
Comment (noamtwix) [reply packages/ricos-editor/web/src/RicosEditor.tsx]: fixed

File: packages/ricos-editor/web/src/RicosEditor.tsx (hunk 12 above)
Comment (noamtwix) [reply packages/ricos-editor/web/src/RicosEditor.tsx]: fixed

File: packages/ricos-editor/web/src/RicosEditor.tsx (hunk 11 above)
Comment (noamtwix) [reply packages/ricos-editor/web/src/RicosEditor.tsx]: fixed

File: examples/main/package.json [hunk 15]
```diff
@@ -58,7 +58,8 @@
     "wix-rich-content-plugin-video": "8.50.2",
     "wix-rich-content-preview": "8.50.2",
     "wix-rich-content-text-selection-toolbar": "8.50.2",
-    "wix-rich-content-ui-components": "8.50.2"
+    "wix-rich-content-ui-components": "8.50.2",
+    "wix-rich-content-toolbars-new": "8.50.2"
```
Comment (shaulgo) [inline examples/main/package.json:62]: we can't have changes in the example app

File: examples/main/shared/editor/Editor.tsx [hunk 16]
```diff
@@ -181,10 +181,11 @@ export default class Editor extends PureComponent<ExampleEditorProps> {
             placeholder={'Add some text!'}
             plugins={this.ricosPlugins}
             linkPanelSettings={{ ...(Plugins.uiSettings.linkPanel || {}), externalPopups }}
+            _rcProps={{ helpers: this.helpers }}
```
Comment (shaulgo) [inline examples/main/shared/editor/Editor.tsx:184]: also needs to remove

File: examples/main/package.json (hunk 15 above)
Comment (noamtwix) [reply examples/main/package.json]: fixed

File: examples/main/shared/editor/Editor.tsx (hunk 16 above)
Comment (noamtwix) [reply examples/main/shared/editor/Editor.tsx:184]: fixed

**Pull request merged**